import "goby/common/protobuf/option_extensions.proto";

package goby.acomms.protobuf;

message NetworkAck
{
  option (goby.msg).dccl.id = 6;
  option (goby.msg).dccl.max_bytes = 32;
  option (goby.msg).queue.ack = false;
  
  required int32 message_src = 1 [(goby.field).dccl.min = -1,
                                  (goby.field).dccl.max = 30,
                                  (goby.field).queue.is_dest=true]; 
  required int32 message_dest = 2 [(goby.field).dccl.min = -1,
                                   (goby.field).dccl.max = 30];
  optional uint64 message_time = 3 [(goby.field).dccl.codec = "_time",
                            (goby.field).queue.is_time=true];

  required int32 message_dccl_id = 4 [(goby.field).dccl.min = 0,
                                      (goby.field).dccl.max = 0x7FFF];

  required int32 ack_src = 10 [(goby.field).dccl.min = -1,
                               (goby.field).dccl.max = 30,
                               (goby.field).queue.is_src=true];
  

}
